### PR TITLE
Jared/3samps basemapstyle

### PR DESCRIPTION
**Pull request by jared-2016**
@ldanzinger @tanneryould Ready for review. Thanks

This PR updates the following 3 samples, Change basemap, Display a map, and Display a Scene to use the new BasemapStyle's. Readmes have been updated where needed as well as the metadata. Screenshots were checked and don't need updating as far as I could tell.

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ChangeBasemap/main.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ int main(int argc, char *argv[])
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ChangeBasemap/main.cpp:42]: Are we missing the code to set the API key?

File: ArcGISRuntimeSDKQt_CppSamples/Maps/DisplayMap/main.cpp [hunk 2]
```diff
@@ -48,7 +48,7 @@ int main(int argc, char *argv[])
 
   // Add the import Path
   view.engine()->addImportPath(QDir(QCoreApplication::applicationDirPath()).filePath("qml"));
-
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/DisplayMap/main.cpp:51]: Again here, should we provide an example of how to set the api key?

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/BasicSceneView.cpp [hunk 3]
```diff
@@ -47,10 +47,9 @@ void BasicSceneView::componentComplete()
   // find QML SceneView component
   m_sceneView = findChild<SceneQuickView*>("sceneView");
 
-  // create a new basemap instance
-  Basemap* basemap = Basemap::imagery(this);
-  // create a new scene instance
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/BasicSceneView.cpp:52]: Should we leave this comment in?

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/main.cpp [hunk 4]
```diff
@@ -55,7 +55,7 @@ int main(int argc, char *argv[])
 
   // Add the import Path
   view.engine()->addImportPath(QDir(QCoreApplication::applicationDirPath()).filePath("qml"));
-
```
Comment (tanneryould) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/main.cpp:58]: Here again

**Review by tanneryould — COMMENTED** [review]
Apart from the code to demonstrate how to set an API key it looks good to me.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ChangeBasemap/ChangeBasemap.cpp [hunk 5]
```diff
@@ -64,31 +64,29 @@ void ChangeBasemap::changeBasemap(const QString& basemap)
...
-      m_map->setBasemap(Basemap::lightGrayCanvas(this));
-    else if (basemap == "Light Gray Canvas (Vector)")
-      m_map->setBasemap(Basemap::lightGrayCanvasVector(this));
-    else if (basemap == "Navigation (Vector)")
-      m_map->setBasemap(Basemap::navigationVector(this));
-    else if (basemap == "OpenStreetMap (Raster)")
-      m_map->setBasemap(Basemap::openStreetMap(this));
+      m_map->setBasemap(new Basemap(BasemapStyle::ArcGISStreets, this));
+    else if (basemap == "Streets - Relief")
+      m_map->setBasemap(new Basemap(BasemapStyle::ArcGISStreetsRelief));
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ChangeBasemap/ChangeBasemap.cpp:71]: missing qobject parent

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ChangeBasemap/ChangeBasemap.cpp [hunk 6]
```diff
@@ -64,31 +64,29 @@ void ChangeBasemap::changeBasemap(const QString& basemap)
...
+    else if (basemap == "Imagery with Labels")
+      m_map->setBasemap(new Basemap(BasemapStyle::ArcGISImagery, this));
+    else if (basemap == "Labels without Imagery")
+      m_map->setBasemap(new Basemap(BasemapStyle::ArcGISImageryLabels, this));
+    else if (basemap == "Dark Gray Canvas")
+      m_map->setBasemap(new Basemap(BasemapStyle::ArcGISDarkGray));
+    else if (basemap == "Light Gray Canvas")
+      m_map->setBasemap(new Basemap(BasemapStyle::ArcGISLightGray));
+    else if (basemap == "Navigation")
+      m_map->setBasemap(new Basemap(BasemapStyle::ArcGISNavigation));
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ChangeBasemap/ChangeBasemap.cpp:85]: missing parent

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/README.md [hunk 7]
```diff
@@ -14,7 +14,7 @@ When loaded, the sample will display a scene. Pan and zoom to explore the scene.
 
 ## How it works
 
-1. Create an `Scene` object with the `Basemap::imagery()` basemap.
+1. Create an `Scene` object with a basemap using the `BasemapStyle::ArcGISImagery` BasemapStyle.
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/README.md:17]:
```suggestion
1. Create a `Scene` object with a basemap using the `BasemapStyle::ArcGISImagery` BasemapStyle.
```

**Review by ldanzinger — CHANGES_REQUESTED** [review]

**Review by tanneryould — COMMENTED** [review]
Added requisite parents

**Review by ldanzinger — APPROVED** [review]
LGTM - the main.cpp addition can happen later. This will allow it to work with the viewer before our final build

**Comment by tanneryould** [comment]
For some reason the github action checks aren't running today. I wonder if it's something to do with GitHub's end? Nothing has changed on our end.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ChangeBasemap/main.cpp (hunk 1 above)
Comment (jared-2016) [reply ArcGISRuntimeSDKQt_CppSamples/Maps/ChangeBasemap/main.cpp:42]: Saving for a later date as it's not necessary atm.

**Comment by ldanzinger** [comment]
@tanneryould - https://www.githubstatus.com/ 

> Investigating - We are investigating reports of degraded performance for GitHub Actions.
Dec 1, 16:40 UTC

File: ArcGISRuntimeSDKQt_CppSamples/Maps/DisplayMap/main.cpp (hunk 2 above)
Comment (jared-2016) [reply ArcGISRuntimeSDKQt_CppSamples/Maps/DisplayMap/main.cpp:51]: same as above.

File: ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/BasicSceneView.cpp (hunk 3 above)
Comment (jared-2016) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/BasicSceneView.cpp:52]: Yeah good catch.

Comment (tanneryould) [reply ArcGISRuntimeSDKQt_CppSamples/Scenes/BasicSceneView/BasicSceneView.cpp:52]: @jared-2016 on it

**Pull request merged**